src/platforms/web/compiler/modules/class.js

```diff
@@ -22,7 +22,7 @@ function transformNode (el: ASTElement, options: CompilerOptions) {
     }
   }
   if (staticClass) {
-    el.staticClass = JSON.stringify(staticClass)
+    el.staticClass = JSON.stringify(staticClass.replace(/\n/g, ' ').replace(/ +/g, ' '))
```

Comment:
would a `/\s+/g` be better? I was looking into this issue as well

Reply:
Good catch - updated